File: backend/src/main/java/tko/model/dto/workout/ExerciseDTO.java
package tko.model.dto.workout;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ExerciseDTO {
    private Long id;
    @NotBlank
    private String instruction;
    @NotBlank
    private String name;
    @NotBlank
    private String difficult;
    @NotBlank
    private String muscularGroup;
    private Integer likeCount;
    @NotNull
    private Boolean requiresEquipment;
    @NotBlank
    private String type;
}

File: app/src/main/java/tk/ssau/fitnesstko/model/dto/ExerciseDto.kt
package tk.ssau.fitnesstko.model.dto

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * Полное представление упражнения.
 * @param id Уникальный идентификатор упражнения (может быть null для новых объектов).
 * @param instruction Подробное описание техники выполнения упражнения.
 * @param name Название упражнения (например, "Приседания со штангой").
 * @param difficult Уровень сложности. Допустимые значения: "EASY", "MEDIUM", "HARD".
 * @param muscularGroup Целевая группа мышц (например, "Ноги", "Грудь", "Спина").
 * @param likeCount Количество лайков. Может быть null, если данные не загружены.
 * @param requestEquipment Требуется ли специальное оборудование (true/false).
 * @param type Тип упражнения (например, "Силовое", "Кардио", "Растяжка").
 */
@Parcelize
data class ExerciseDto(
    val id: Long?,
    val instruction: String,
    val name: String,
    val difficult: String,
    val muscularGroup: String,
    val likeCount: Int?,
    val requestEquipment: Boolean,
    val type: String
) : Parcelable
